### PR TITLE
WIP. Move image build step to 'deploy' phase on Travis CI

### PR DESCRIPTION
**Pull request by e0ne**
We'll build image on Jenkins on each pull request.

Travis CI will build image only during deploy phase to not face
DockerHub image pull limit.

depends-on: kubernetes-ci#131

### CODE REVIEW

File: example/rdma-test-pod1.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: v1
+ apiVersion: v1
```
Comment (moshe010) [inline example/rdma-test-pod1.yml:1]: why this change?

File: .travis.yml [hunk 2]
```diff
@@ -31,7 +30,9 @@ deploy:
   # Push image to Dockerhub on tag
   - provider: script
     skip_cleanup: true
-    script: bash scripts/docker-push.sh $IMAGE_NAME $TRAVIS_TAG $TRAVIS_CPU_ARCH
+    script:
+      - make image
```
Comment (moshe010) [inline .travis.yml:34]: how will we know that the image build is passing before merging PR?

Comment (e0ne) [reply .travis.yml:34]: It will be tested in https://github.com/Mellanox/kubernetes-ci/pull/131

**Comment by moshe010** [comment]
I think we agree that this is not needed
